### PR TITLE
network: several cleanups for reloading .network files

### PR DESCRIPTION
**Pull request by yuwata**
split-out of #34909.

### CODE REVIEW

File: src/network/netdev/netdev.c [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "networkd-manager.h"
 #include "networkd-queue.h"
 #include "networkd-setlink.h"
+#include "networkd-state-file.h"
 #include "networkd-sriov.h"
```
Comment (poettering) [inline src/network/netdev/netdev.c:36]: not sorted?

File: src/network/netdev/netdev.c [hunk 2]
```diff
@@ -995,15 +995,12 @@ int netdev_load_one(Manager *manager, const char *filename) {
         return 0;
```
Comment (poettering) [inline src/network/netdev/netdev.c:1014]: something went wrong with the commit msg? the first and 2nd line seem duplicative?

File: src/network/netdev/netdev.c [hunk 3]
```diff
@@ -995,15 +995,12 @@ int netdev_load_one(Manager *manager, const char *filename) {
...
 
-int netdev_load(Manager *manager, bool reload) {
+int netdev_load(Manager *manager) {
         _cleanup_strv_free_ char **files = NULL;
         int r;
 
         assert(manager);
 
-        if (!reload)
-                hashmap_clear_with_destructor(manager->netdevs, netdev_unref);
```
Comment (poettering) [inline src/network/netdev/netdev.c:1013]: maybe add an `assert(hashmap_isempty(…` instead?

File: src/network/netdev/netdev.c (hunk 2 above)
Comment (yuwata) [reply src/network/netdev/netdev.c:1014]: I do not think so? Explained two functions. I will rephrase later.

File: src/network/netdev/netdev.c (hunk 3 above)
Comment (yuwata) [reply src/network/netdev/netdev.c:1013]: No, when this is called on reload, the hashmap is _not_ empty.

**Comment by yuwata** [comment]
@poettering Thank you for the review. Addressed all comments, except for the last one. Upgrading the green label.

**Pull request merged**